tests/ethereum/test_general.py

```diff
@@ -1793,6 +1793,83 @@ def test_FilterFunctions_fallback_function_matching(self):
                 ABI.deserialize("uint", to_constant(m.world.transactions[-1].return_data)), 456
             )
 
+    def test_checkpoint(self):
+        #test enable/disable plugin and sync vs contextmanager
+        source_code = """
+        contract C {
+            constructor() public payable {}
+            function f1(uint a) public payable {}
+            function f2(uint a) public payable {}
+        }
+        """
+
+        m: ManticoreEVM = ManticoreEVM()
+
+        creator_account = m.create_account(balance=10000000000)
+        contract_account = m.solidity_create_contract(source_code, owner=creator_account, balance=0)
+
+        #Can not go to unexistant snapshot
+        self.assertRaises(Exception, m.goto_snapshot)
```

Comment:
We can make this `ManticoreError` to match the behavior in `only_from_main_script`